### PR TITLE
implement 'cl write <target> <string>'

### LINKED ISSUE #199: interaction with bundles
Summary: Be able to interact with your run bundles when they are executing by writing files that your process can read.

1) An example of the general usage:

```
cl write bundle_uuid/commands.txt "set eta 0.5"
```

Your program will be responsible for reading commands.txt and interpreting what's there.

2) A common use case is to prepend to your program with a standard script (which we can provide) to read commands and just executes them:

```
cl run './get_commands & your_command'
```

Then you can execute arbitrary commands in the same docker container as your main program:

```
cl write bundle_uuid/command.txt "grep error log.txt"
```

3) To have an interactive prompt, in your program, do this:

```
mkfifo stdin
sleep 100000 > stdin &
your_program < stdin
```

Then from the outside, you can write to the pipe.

```
cl write bundle_uuid/stdin "hello"
```

Implementation: This should be implemented by the 'actions' metadata field (handles 'kill' right now).
Need to record the time that the actions were carried out.

Move 'bundle_metadata / actions' into bundle_action, and add fields to keep track of time.

### PR DESCRIPTION
**Pull request by percyliang**
Fixes #199 
@kashizui please review

### CODE REVIEW

File: codalab/machines/local_machine.py [hunk 1]
```diff
@@ -59,10 +60,13 @@ def start_bundle(self, bundle, bundle_store, parent_dict, username):
...
-    def kill_bundle(self, bundle):
+    def send_bundle_action(self, bundle, action_string):
         if not self.bundle or self.bundle.uuid != bundle.uuid: return False
-        print >>sys.stderr, 'LocalMachine.kill_bundle %s' % (bundle.uuid)
-        self.process.kill()
+        print >>sys.stderr, 'LocalMachine.send_command(%s, %s)' % (bundle.uuid, action_string)
+        if action_string == BundleAction.KILL:
+            self.process.kill()
+        else:
+            print '=== Unhandled command: %s' % action_string
```
Comment (kashizui) [inline codalab/machines/local_machine.py]: nit: maybe `>>stderr`?

File: codalab/lib/bundle_action.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class BundleAction(object):
+    KILL = 'kill'
+    WRITE = 'write'
+
+    """
```
Comment (kashizui) [inline codalab/lib/bundle_action.py:2]: nit: docstring should be on first line

File: codalab/lib/bundle_action.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    A bundle action is serialized as a string, which consists of a sequence of arguments.
+    """
+
+    @staticmethod
+    def kill():
+        return BundleAction.KILL
+
+    @staticmethod
+    def write(subpath, string):
+        return ' '.join([BundleAction.WRITE, subpath, string])
```
Comment (kashizui) [inline codalab/lib/bundle_action.py]: Will there be any issues with subpaths with spaces?

**Comment by kashizui** [comment]
LGTM modulo comments

File: codalab/lib/bundle_action.py (hunk 3 above)
Comment (percyliang) [inline codalab/lib/bundle_action.py]: Good point - will enforce that subpath isn't arbitrary

**Pull request merged**